File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.py
import sys

n = int(sys.stdin.readline())
emoji = set()
cnt = 0
for _ in range(n):
    s = sys.stdin.readline().rstrip()
    if s != 'ENTER':
        if s not in emoji:
            emoji.add(s)
            cnt += 1
    elif s == 'ENTER':
        emoji.clear()
print(cnt)

File: 백준/Silver/25192. 인사성 밝은 곰곰이/인사성 밝은 곰곰이.cc
#include <iostream>
#include <string>
#include <set>
using namespace std;

int main() {
	cin.tie(0)->sync_with_stdio(0);

	int n, cnt = 0;
	set<string> s;
	string str;
	cin >> n;
	while (n--) {
		cin >> str;
		if (str == "ENTER") {
			cnt += s.size();
			s.clear();
			continue;
		}
		s.emplace(str);
	}
	cnt += s.size();
	cout << cnt;
}